src/h3/mod.rs

```diff
@@ -657,6 +682,20 @@ impl Connection {
         // Return how many bytes were written, excluding the frame header.
         let written = conn.stream_send(stream_id, body, fin)?;
 
+        // tidy up streams
+        if !fin {
+            return Ok(written);
+        }
+
+        if self
```

Comment:
Try instead something like:

```rust
match self.streams.entry(stream_id) {
  Entry::Occupied(entry) if entry.get().peer_fin() => {
    entry.remove();
  }
  _ => {}
}
```

Reply:
Oh? What failed?

Reply:
Compiler complained that Entry was unknown. A tried a few things but gave up.

Reply:
Unknown as in, you just need to `use ...` it to get in scope first?